packages/calcite-components/src/utils/locale.ts

```diff
@@ -175,6 +175,23 @@ export function getSupportedLocale(locale: string, context: "cldr" | "t9n" = "cl
   return locale;
 }
 
+/**
+ * Gets the locale that best matches the context for date formatting.
```

Comment:
Could you add a bit of info for this workaround? Linking the issue could also be helpful.